pkg/controllers/scheduler/scheduler.go

```diff
@@ -361,6 +361,39 @@ func (s *Scheduler) prepareToSchedule(
 ) {
 	logger := klog.FromContext(ctx)
 
+	// check migration config
+
+	existingMigrationConfigBytes, migrationConfigExists := fedObject.GetAnnotations()[common.MigrationConfigurationAnnotation]
```

Comment:
We need to place it here because when applied migration config does not change we do not trigger re-schedule. Therefore parsing should be before the logic to check whether to trigger scheduling.

Reply:
Understood. This looks like a gotcha. Can we add a comment to explain the above for future developers so they do not accidentally move it to `applySchedulingResults`?

Reply:
done.